### PR TITLE
Refactor command-line scripts

### LINKED ISSUE #91: Rename scripts to be Euphonic specific
Euphonic's utility scripts (e.g. `dispersion.py`, `dos.py`) have very generic names which could easily lead to clashes on a user's path. They should be renamed (e.g. `euphonic-dispersion`) and the old names deprecated so the old names can be removed in a future version.

### PR DESCRIPTION
**Pull request by ajjackson**
Closes #91 

Move scripts to a submodule of Euphonic for cleaner access/testing/distribution.

This allows them to be installed with command-line-friendly aliases e.g. `euphonic-optimise-eta`. If you have a developer (`-e`) install of Euphonic, you'll need to re-rerun `pip install` to get these installed into your local environment. New users shouldn't notice much change apart from the improved names.

### CODE REVIEW

File: doc/source/disp-script.rst [hunk 1]
```diff
@@ -20,7 +20,6 @@ Command Line Options
 --------------------
 
 .. argparse::
-   :filename: ../../scripts/dispersion.py
+   :module: euphonic.cli.dispersion
```
Comment (rebeccafair) [inline doc/source/disp-script.rst:23]: All instances of `dispersion.py` should be changed to `euphonic-dispersion` in this file

File: doc/source/tutorials.rst [hunk 2]
```diff
@@ -32,6 +32,7 @@ Helper Scripts
 - :ref:`dispersion.py <disp-script>`
 - :ref:`dos.py <dos-script>`
 - :ref:`optimise_eta.py <eta-script>`
```
Comment (rebeccafair) [inline doc/source/tutorials.rst:34]: These should also be updated to `euphonic-dispersion` etc.

**Review by rebeccafair — COMMENTED** [review]
Looks good! Just a few remaining references I found with grep. Provided the docs build ok once the cli package is added, if you could add an entry to CHANGELOG.rst then we're all good :)

File: doc/source/dos-script.rst [hunk 3]
```diff
@@ -20,6 +20,6 @@ Command Line Options
 --------------------
 
 .. argparse::
-   :filename: ../../scripts/dos.py
+   :module: euphonic.cli.dos
```
Comment (rebeccafair) [inline doc/source/dos-script.rst:23]: References to `dos.py` in this file also need to be updated to `euphonic-dos`

File: setup.py [hunk 4]
```diff
@@ -65,11 +65,6 @@ def run_setup(build_c=True):
                 'euphonic.readers',
                 'euphonic.data']
```
[the remark this answers is not in the document]
Comment (ajjackson) [reply setup.py:66]: Ok, can do. I think in most setup.py files this is covered by some kind of auto-discovery function.

File: doc/source/disp-script.rst (hunk 1 above)
Comment (ajjackson) [reply doc/source/disp-script.rst:23]: done!

File: doc/source/tutorials.rst (hunk 2 above)
Comment (ajjackson) [reply doc/source/tutorials.rst]: done

File: doc/source/dos-script.rst (hunk 3 above)
Comment (ajjackson) [reply doc/source/dos-script.rst:23]: done

**Review by rebeccafair — CHANGES_REQUESTED** [review]
Just 2 more things I spotted!

Could you also add an entry to the changelog?

File: doc/source/eta-script.rst [hunk 5]
```diff
@@ -67,14 +67,14 @@ reciprocal sums.
...
+   euphonic-optimise-eta -h
 
 Or see the command line options in more detail below
 
 Command Line Options
 --------------------
 
 .. argparse::
-   :filename: ../../scripts/optimise_eta.py
+   :filename: euphonic.cli.optimise_eta
```
[the remark this answers is not in the document]
Comment (ajjackson) [reply doc/source/eta-script.rst]: done!

File: doc/source/dos-script.rst [hunk 6]
```diff
@@ -1,25 +1,25 @@
...
-=============
+============
+euphonic-dos
+============
 
-``dos.py`` can be used to plot a density of states from precalculated
-frequencies in a ``.phonon`` or ``.bands`` file. For example, to plot DOS from a
-``quartz.phonon`` file, run::
+The ``euphonic-dos`` program can be used to plot a density of states
+from precalculated frequencies in a ``.phonon`` or ``.bands``
```
[the remark this answers is not in the document]
Comment (ajjackson) [reply doc/source/dos-script.rst]: done!

**Review by rebeccafair — APPROVED** [review]

**Pull request merged**